azurerm/internal/services/cosmos/resource_arm_cosmosdb_mongo_collection.go

```diff
@@ -321,24 +384,59 @@ func expandCosmosMongoCollectionIndexes(defaultTtl *int) *[]documentdb.MongoInde
 		})
 	}
 
-	return &outputs
+	return &results
 }
 
-func flattenCosmosMongoCollectionIndexes(indexes *[]documentdb.MongoIndex) *int {
-	var ttl int
-	for _, i := range *indexes {
-		if key := i.Key; key != nil {
-			var ttlInner int32
+func flattenCosmosMongoCollectionIndex(input *[]documentdb.MongoIndex) (*[]map[string]interface{}, *[]map[string]interface{}, *int32) {
+	indexes := make([]map[string]interface{}, 0)
+	systemIndexes := make([]map[string]interface{}, 0)
+	var ttl *int32
+
+	for _, v := range *input {
+		index := map[string]interface{}{}
+		systemIndex := map[string]interface{}{}
+
+		if v.Key != nil && v.Key.Keys != nil {
+			key := (*v.Key.Keys)[0]
```

Comment:
there's a crash here if there's 0 items in this

Reply:
it always wouldn't be 0 since api always return system key "_id".